MericCelik/assignment1/Q1.java

```diff
@@ -0,0 +1,202 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class Q1 {
+
+    /*The function checks whether two strings are anagram. It is case insensitive and it discards all
+    * characters other than letters. The logic behind it is calculating character's ascii values and
+    * using them as an index of arrays. It stores the frequency of letters of original string and compares
+    * it with resulting string.*/
+
+    public static boolean isAnagram(String original, String resulting){
```

Comment:
optional: You've provided a few implementations for different cases, which is very nice. One thing that is noticeable though is that there is a bit of code duplication here. Do you think you could refactor it to reduce the duplication?

Reply:
I merged two isAnagram() functions by using isSensitive parameter. It looks better I guess. :blush: I am trying to implement all different solutions to compare them and inspect the problem from different perspectives. Do you recommend me to continue doing this for future assignment?

Reply:
It's enough to provide one solution. :) This anagram assignment was a little verbose, and you could choose to accept different types of anagrams, that's where the multiple answers came from. In the future, let's focus on one answer - for me, here the method isAnagram_fullAscii is extensive enough and I would accept it even without the other methods.
If you have any questions about future assignments like what's the input, expected output, what functionality needs to be implemented - please message me, so you don't have to spend so much time again finding the super-set of all solutions. :)
However, it is very good thinking that you have shown here and it will be very useful during your interviews - just try to ask questions first and narrow down the solution as much as possible before you jump into implementation. It is important to see different solutions, but it is also important to just choose the best one.

Reply:
I will keep that in mind, thank you!